extensions/mssql/src/sqlClusterLookUp.ts

```diff
@@ -0,0 +1,147 @@
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as sqlops from 'sqlops';
+import * as constants from './constants';
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import { AppContext } from './appContext';
+import { SqlClusterConnection } from './objectExplorerNodeProvider/connection';
+import { ICommandObjectExplorerContext, ICommandUnknownContext, ICommandViewContext } from './objectExplorerNodeProvider/command';
+import { SqlObjectExplorerNodeProvider } from './objectExplorerNodeProvider/objectExplorerNodeProvider';
+
+export class SqlClusterLookUp {
+	public static findSqlClusterConnection(
+		obj: ICommandObjectExplorerContext | sqlops.IConnectionProfile,
+		appContext: AppContext) : SqlClusterConnection  {
+
+		if (!obj || !appContext) { return undefined; }
+
+		let sqlConnProfile: sqlops.IConnectionProfile;
+		if ('type' in obj && obj.type === constants.ObjectExplorerService
+			&& 'explorerContext' in obj && obj.explorerContext && obj.explorerContext.connectionProfile) {
+			sqlConnProfile = obj.explorerContext.connectionProfile;
+		} else if ('options' in obj) {
+			sqlConnProfile = obj;
+		}
+
+		let sqlClusterConnection: SqlClusterConnection = undefined;
+		if (sqlConnProfile) {
+			sqlClusterConnection = this.findSqlClusterConnectionBySqlConnProfile(sqlConnProfile, appContext);
+		}
+		return sqlClusterConnection;
+	}
+
+	private static findSqlClusterConnectionBySqlConnProfile(sqlConnProfile: sqlops.IConnectionProfile, appContext: AppContext): SqlClusterConnection {
+		if (!sqlConnProfile || !appContext) { return undefined; }
+
+		let sqlOeNodeProvider = appContext.getService<SqlObjectExplorerNodeProvider>(constants.ObjectExplorerService);
+		if (!sqlOeNodeProvider) { return undefined; }
+
+		let sqlClusterSession = sqlOeNodeProvider.findSqlClusterSessionBySqlConnProfile(sqlConnProfile);
+		if (!sqlClusterSession) { return undefined; }
+
+		return  sqlClusterSession.sqlClusterConnection;
+	}
+
+	public static async getSqlClusterConnInfo(
+		obj: sqlops.IConnectionProfile | sqlops.connection.Connection | ICommandObjectExplorerContext): Promise<ConnectionParam> {
+
+		if (!obj) { return undefined; }
+
+		let sqlClusterConnInfo: ConnectionParam = undefined;
+		if ('providerName' in obj) {
+			if (obj.providerName === constants.mssqlClusterProviderName) {
+				sqlClusterConnInfo = 'id' in obj ? this.connProfileToConnectionParam(obj) : this.connToConnectionParam(obj);
+			} else {
+				sqlClusterConnInfo = await this.createSqlClusterConnInfo(obj);
+			}
+		} else {
+			sqlClusterConnInfo = await this.createSqlClusterConnInfo(obj.explorerContext.connectionProfile);
+		}
+
+		return sqlClusterConnInfo;
+	}
+
+	private static async createSqlClusterConnInfo(sqlConnInfo: sqlops.IConnectionProfile | sqlops.connection.Connection): Promise<ConnectionParam> {
```

Comment:
As mentioned elsewhere do we really need this? Can't we just go straight to `SqlClusterConnection` class now instead of doing convert from SQL -> Knox as a ConnectionInfo object?

Reply:
Yes. This method is being used at `MssqlSqlObjectExplorerNodeProvider.doSessionOpen()`